### PR TITLE
Fix cloud-stream-kafka sample for 0.11.x

### PR DESCRIPTION
**Pull request by sobychacko**
With the changes in this PR, the sample in `cloud-stream-kafka` is working on `0.11.x` branch for the command: `build.sh --aot-only`. 

However, it fails for the non-aot mode build with `build.sh`.  Do you know why there is a difference between the two modes of build execution?

### CODE REVIEW

**Comment by snicoll** [comment]
`build.sh` builds with AOT, create a native image and then check that the output is what's expected using `verifh.sh`.

`build.sh --aot-only` builds with AOT, start the app on the JVM and then check that the output is what's expected using `verifh.sh`.

So if it's failing with the former, some hints for GraalVM are missing? To be clear both executions uses the AOT engine (on native or on the JVM/hotspot respectively).

File: spring-native-configuration/src/main/java/org/springframework/cloud/stream/kafka/KafkaBinderHints.java [hunk 1]
```diff
@@ -33,7 +34,7 @@
 						KafkaBinderHealthIndicatorConfiguration.class,
 						KafkaExtendedBindingProperties.class,
 						KafkaBindingProperties.class
-				})
+				}, access = AccessBits.FULL_REFLECTION)
```
Comment (sdeleuze) [inline spring-native-configuration/src/main/java/org/springframework/cloud/stream/kafka/KafkaBinderHints.java:37]: Could you please elaborate on why this change is required?

**Comment by sdeleuze** [comment]
Closing this PR as cloud-stream-kafka sample is using dynamic application context and can't be supported on 0.11.
